src/class.c

```diff
@@ -767,47 +771,132 @@ boot_defclass(mrb_state *mrb, struct RClass *super)
   return c;
 }
 
-MRB_API void
-mrb_include_module(mrb_state *mrb, struct RClass *c, struct RClass *m)
+static void
+boot_initmod(mrb_state *mrb, struct RClass *mod)
 {
-  struct RClass *ins_pos;
+  mod->origin = mod;
+  mod->mt = kh_init(mt, mrb);
+}
 
-  ins_pos = c;
-  while (m) {
-    struct RClass *p = c, *ic;
+static struct RClass*
+include_class_new(mrb_state *mrb, struct RClass *m, struct RClass *super)
```

Comment:
It moves the noisy `ICLASS` creation into a function, this code is very distracting when placed into  `include_module_at`, since its already a complicated function, try describing the function without using too many `and` and you'll see what I mean.

tl;dr separation of concerns